packages/jmix-front-generator/src/building-blocks/stages/writing/pieces/previews-registration.ts

```diff
@@ -0,0 +1,30 @@
+import {registerComponentPreviews, addComponent, RouteInfo} from "../../../../generators/react-typescript/common/previews-registartion";
+import {YeomanGenerator} from "../../../YeomanGenerator";
+
+export function addComponentPreviews(
+  gen: YeomanGenerator,
+  dirShift: string,
+  className: string,
+  nameLiteral: string,
+  isDefaultExport: boolean = false,
+  componentProps?: any,
+  registerComponentCallback: (
+    routingContents: string,
+    routeInfo: RouteInfo,
+    componentProps?: any
+  ) => string = addComponent
+) {
+  if (!registerComponentPreviews(gen.fs, {
+      componentFileName: className,
+      componentClassName: className,
+      dirShift: dirShift,
+      destRoot: gen.destinationRoot(),
+      pathPattern: '/' + nameLiteral
```

Comment:
It's a fragile since there may be several components with the same name. ideally we should add path or some random salt